contracts/src/CrossChainApplications/GETTING_STARTED.md

```diff
@@ -4,31 +4,70 @@ This section walks through how to build an example cross-chain application on to
 
 ## Step 1: Create Initial Contract
 
-Create a new file called `MyExampleCrossChainMessenger.sol` in the directory that will hold the application.
+Create a new file called `MyExampleCrossChainMessenger.sol` in a `teleporter/contracts/src/CrossChainApplications/MyExampleCrossChainMessenger/` directory.
 
 At the top of the file define the Solidity version to work with, and import the necessary types and interfaces.
 
 ```solidity
 pragma solidity 0.8.18;
 
 import {ITeleporterMessenger, TeleporterMessageInput, TeleporterFeeInfo} from "../../Teleporter/ITeleporterMessenger.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
 import {ITeleporterReceiver} from "../../Teleporter/ITeleporterReceiver.sol";
 ```
 
-Next, define the initial empty contract.
+Next, define the initial empty contract. The contract inherits from `ReentrancyGuard` to prevent reentrancy attacks, and inherits from `ITeleporterReceiver` to allow the contract to receive messages from Teleporter.
 
 ```solidity
-contract MyExampleCrossChainMessenger {}
+contract MyExampleCrossChainMessenger is
+    ReentrancyGuard,
+    ITeleporterReceiver
+{}
+```
+
+Finally, add the following struct and event declarations to the contract, which will be integrated in later:
```

Comment:
```suggestion
Add the following struct and event declarations to the contract, which will be integrated in later:
```

Reply:
Gonna keep this as is to match the flow of the rest of this step.